### PR TITLE
Requested changes for integration in Thunderbird

### PR DESCRIPTION
**Pull request by leonbotros**
First commit is organisational. The second commit adds small changes needed for thunderbird.

### CODE REVIEW

**Review by Iso5786 — CHANGES_REQUESTED** [review]
See my 2 comments.

File: src/composeMail.ts [hunk 1]
```diff
@@ -51,50 +51,44 @@ export class ComposeMail implements IComposeIrmaSealMail {
...
-                'Content-Type': `multipart/encrypted; protocol="application/irmaseal"; boundary=${this.boundary}`,
-            }
-
-            let headerStr = ''
-            for (const [k, v] of Object.entries(headers)) {
-                headerStr += `${k}: ${v}\r\n`
-            }
-            headerStr += '\r\n\r\n'
-            return headerStr
+        if (!this.ct) throw new Error('No ciphertext')
```
[the remark this answers is not in the document]
Comment (leonbotros) [reply src/composeMail.ts:77]: The CT check was already there. Thunderbird already fills in the subject, recipient and sender before passing it off to the mime proxy responsible for encrypting the message.

[the remark this answers is not in the document]
Comment (leonbotros) [reply src/composeMail.ts:77]: But I agree, it can go.

File: src/composeMail.ts [hunk 2]
```diff
@@ -1,8 +1,8 @@
-// generate random string (30 - 40 chars)
+import { Buffer } from 'buffer'
```
[the remark this answers is not in the document]
Comment (leonbotros) [reply src/composeMail.ts:1]: Buffer is required for encoding `Uint32Array` to a b64 string and back. Thunderbird does not have this implementation available natively. We could
a) polyfill if it isn't available,
b) do the encoding/decoding in our respective application (TB/Outlook).

**Pull request closed without merging**